### PR TITLE
media-gfx/tuxpaint: version bump to 0.9.24

### PR DESCRIPTION
**Pull request by crabbedhaloablution**
Closes: https://bugs.gentoo.org/385951
Closes: https://bugs.gentoo.org/581036
Closes: https://bugs.gentoo.org/671528
Closes: https://bugs.gentoo.org/734048
Package-Manager: Portage-3.0.12, Repoman-3.0.2
Signed-off-by: Peter Alfredsen <crabbedhaloablution@icloud.com>

### CODE REVIEW

File: media-gfx/tuxpaint/tuxpaint-0.9.24.ebuild [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit toolchain-funcs xdg-utils desktop
```
Comment (thesamesam) [inline media-gfx/tuxpaint/tuxpaint-0.9.24.ebuild:6]: sort

File: media-gfx/tuxpaint/tuxpaint-0.9.24.ebuild [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+DESCRIPTION="Drawing program designed for young children"
+HOMEPAGE="http://www.tuxpaint.org/"
+SRC_URI="mirror://sourceforge/${PN}/${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~x86"
+IUSE=""
```
Comment (thesamesam) [inline media-gfx/tuxpaint/tuxpaint-0.9.24.ebuild:15]: We can [drop](https://dev.gentoo.org/~ulm/pms/head/pms.html#x1-620007.3) empty IUSE

File: media-gfx/tuxpaint/tuxpaint-0.9.24.ebuild [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	media-libs/sdl-mixer
+	media-libs/sdl-pango
+	media-libs/sdl-ttf
+	x11-libs/cairo
+	"
+DEPEND="${RDEPEND}"
+BDEPEND="sys-devel/gettext
+	media-gfx/imagemagick[png]
+	dev-util/gperf
+	"
```
Comment (thesamesam) [inline media-gfx/tuxpaint/tuxpaint-0.9.24.ebuild:34]: I usually sort these by category then inter-category (like you did for RDEPEND)

File: media-gfx/tuxpaint/tuxpaint-0.9.24.ebuild [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+pkg_postinst() {
+	xdg_icon_cache_update
+
+	if ! has_version "${CATEGORY}/${PN}"; then
+		elog "For additional graphic stamps, you can emerge the"
+		elog "media-gfx/tuxpaint-stamps package."
+	fi
+}
+
+pkg_postrm() {
```
Comment (thesamesam) [inline media-gfx/tuxpaint/tuxpaint-0.9.24.ebuild:76]: It's completely valid to do this with xdg-utils.eclass, but let's use the nicer xdg.eclass with its [autologic](https://gitweb.gentoo.org/repo/gentoo.git/tree/eclass/xdg.eclass?id=e543d9d8a52e29dd939f425ba6d947754f4b59b4#n68).

File: media-gfx/tuxpaint/tuxpaint-0.9.24.ebuild [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+BDEPEND="sys-devel/gettext
+	media-gfx/imagemagick[png]
+	dev-util/gperf
+	"
+
+src_prepare() {
+	default
+	sed -i \
+		-e 's|linux_ARCH_INSTALL:=install-xdg|linux_ARCH_INSTALL:=|' \
+		-e "s|linux_PREFIX:=/usr/local|linux_PREFIX:=/usr|" \
```
Comment (thesamesam) [inline media-gfx/tuxpaint/tuxpaint-0.9.24.ebuild:40]: I wonder if we need to split in a ${EPREFIX} before the /usr, actually.

File: media-gfx/tuxpaint/tuxpaint-0.9.24.ebuild [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+DESCRIPTION="Drawing program designed for young children"
+HOMEPAGE="http://www.tuxpaint.org/"
+SRC_URI="mirror://sourceforge/${PN}/${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~x86"
+IUSE=""
+
+RDEPEND="
```
Comment (thesamesam) [inline media-gfx/tuxpaint/tuxpaint-0.9.24.ebuild:17]: I haven't checked these but it's an important point to make (so why not do it now?) - if we're linking against any libraries and they provide a [subslot](https://devmanual.gentoo.org/general-concepts/slotting/), we should subscribe to it with :=.

File: media-gfx/tuxpaint/tuxpaint-0.9.24.ebuild [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+	newmenu src/tuxpaint.desktop tux4kids-tuxpaint.desktop
+	dodoc docs/*.txt
+	dodoc docs/en/*.txt
+}
+
+pkg_postinst() {
+	xdg_icon_cache_update
+
+	if ! has_version "${CATEGORY}/${PN}"; then
+		elog "For additional graphic stamps, you can emerge the"
```
Comment (thesamesam) [inline media-gfx/tuxpaint/tuxpaint-0.9.24.ebuild:71]: You might want [optfeature.eclass](https://devmanual.gentoo.org/eclass-reference/optfeature.eclass/index.html)!

Comment (crabbedhaloablution) [reply media-gfx/tuxpaint/tuxpaint-0.9.24.ebuild:71]: optfeature is a bit more suited to something that has more than one optional dependency, I think. I will tweak the logic to work like it though.

File: media-gfx/tuxpaint/tuxpaint-0.9.24.ebuild (hunk 5 above)
Comment (crabbedhaloablution) [reply media-gfx/tuxpaint/tuxpaint-0.9.24.ebuild:40]: It'll take more than that to make this PREFIX-aware, I'm afraid. I fiddled with it minimally, and it doesn't quite work, there's always something that doesn't hit the right bucket.

File: media-gfx/tuxpaint/tuxpaint-0.9.24.ebuild (hunk 6 above)
Comment (crabbedhaloablution) [reply media-gfx/tuxpaint/tuxpaint-0.9.24.ebuild:14]: Checked. No subslots.

File: media-gfx/tuxpaint/tuxpaint-0.9.24.ebuild (hunk 4 above)
Comment (crabbedhaloablution) [reply media-gfx/tuxpaint/tuxpaint-0.9.24.ebuild]: Incorporated

File: media-gfx/tuxpaint/tuxpaint-0.9.24.ebuild (hunk 3 above)
Comment (crabbedhaloablution) [reply media-gfx/tuxpaint/tuxpaint-0.9.24.ebuild:32]: Incorporated

File: media-gfx/tuxpaint/tuxpaint-0.9.24.ebuild (hunk 2 above)
Comment (crabbedhaloablution) [reply media-gfx/tuxpaint/tuxpaint-0.9.24.ebuild]: Incorporated

File: media-gfx/tuxpaint/tuxpaint-0.9.24.ebuild (hunk 1 above)
Comment (crabbedhaloablution) [reply media-gfx/tuxpaint/tuxpaint-0.9.24.ebuild]: Incorporated

**Comment by thesamesam** [comment]
I hit a 'convert' error with minimal ImageMagick flags. I added in a bunch (including postscript) and that helped. If you have time, trying with no flags on IM and building it up would be helpful, thanks!

I also had this:
```

 * QA Notice: Files built without respecting LDFLAGS have been detected
 *  Please include the following list of files in your report:
 * /usr/bin/tuxpaint
 * /usr/lib/tuxpaint/plugins/alien.so
 * /usr/lib/tuxpaint/plugins/blind.so
 * /usr/lib/tuxpaint/plugins/blocks_chalk_drip.so
 * /usr/lib/tuxpaint/plugins/blur.so
 * /usr/lib/tuxpaint/plugins/bricks.so
 * /usr/lib/tuxpaint/plugins/calligraphy.so
 * /usr/lib/tuxpaint/plugins/cartoon.so
 * /usr/lib/tuxpaint/plugins/confetti.so
 * /usr/lib/tuxpaint/plugins/distortion.so
 * /usr/lib/tuxpaint/plugins/emboss.so
 * /usr/lib/tuxpaint/plugins/fade_darken.so
 * /usr/lib/tuxpaint/plugins/fisheye.so
 * /usr/lib/tuxpaint/plugins/flower.so
 * /usr/lib/tuxpaint/plugins/foam.so
 * /usr/lib/tuxpaint/plugins/fold.so
 * /usr/lib/tuxpaint/plugins/fretwork.so
 * /usr/lib/tuxpaint/plugins/glasstile.so
 * /usr/lib/tuxpaint/plugins/grass.so
 * /usr/lib/tuxpaint/plugins/halftone.so
 * /usr/lib/tuxpaint/plugins/kalidescope.so
 * /usr/lib/tuxpaint/plugins/light.so
 * /usr/lib/tuxpaint/plugins/metalpaint.so
 * /usr/lib/tuxpaint/plugins/mirror_flip.so
 * /usr/lib/tuxpaint/plugins/mosaic.so
 * /usr/lib/tuxpaint/plugins/mosaic_shaped.so
 * /usr/lib/tuxpaint/plugins/negative.so
 * /usr/lib/tuxpaint/plugins/noise.so
 * /usr/lib/tuxpaint/plugins/perspective.so
 * /usr/lib/tuxpaint/plugins/puzzle.so
 * /usr/lib/tuxpaint/plugins/rails.so
 * /usr/lib/tuxpaint/plugins/rain.so
 * /usr/lib/tuxpaint/plugins/rainbow.so
 * /usr/lib/tuxpaint/plugins/realrainbow.so
 * /usr/lib/tuxpaint/plugins/ripples.so
 * /usr/lib/tuxpaint/plugins/rosette.so
 * /usr/lib/tuxpaint/plugins/sharpen.so
 * /usr/lib/tuxpaint/plugins/shift.so
 * /usr/lib/tuxpaint/plugins/smudge.so
 * /usr/lib/tuxpaint/plugins/snow.so
 * /usr/lib/tuxpaint/plugins/string.so
 * /usr/lib/tuxpaint/plugins/tint.so
 * /usr/lib/tuxpaint/plugins/toothpaste.so
 * /usr/lib/tuxpaint/plugins/tornado.so
 * /usr/lib/tuxpaint/plugins/tv.so
 * /usr/lib/tuxpaint/plugins/waves.so
 * /usr/lib/tuxpaint/plugins/xor.so
```

You can detect these yourself by using:
```
COMMON_FLAGS="-O2 -pipe -march=native -fdiagnostics-color=always -frecord-gcc-switches"
CFLAGS="${COMMON_FLAGS}"
CXXFLAGS="${COMMON_FLAGS}"
FCFLAGS="${COMMON_FLAGS}"
FFLAGS="${COMMON_FLAGS}"
LDFLAGS="${LDFLAGS} -Wl,--defsym=__gentoo_check_ldflags__=0"
PORTAGE_ELOG_CLASSES="${PORTAGE_ELOG_CLASSES} qa"
```

**Pull request closed without merging**